standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -9768,7 +9767,8 @@ private void writeMPartitionColumnStatistics(Table table, Partition partition,
 
       for (ColumnStatisticsObj statsObj : statsObjs) {
         MPartitionColumnStatistics mStatsObj =
-            StatObjectConverter.convertToMPartitionColumnStatistics(mPartition, statsDesc, statsObj, colStats.getEngine());
+            StatObjectConverter.convertToMPartitionColumnStatistics(mPartition, statsDesc, statsObj,
+                colStats.getEngine(), getDefaultCatalog(conf));
```

Comment:
I think the catalog name should be already accessible thru: `MPartition -> MTable -> MDatabase ->catalogName`

Reply:
you are right, fix as suggested, the `MPartition` is nullable here due to https://github.com/apache/hive/blob/baf8247af78f35a68fab0756f9d571c73cdf5f96/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java#L188-L192, fill the statsDesc with the catalog in this case: https://github.com/apache/hive/pull/2910/files#diff-f832bc7a00499f47169832b5ab5339dc879c994bfb955bd0c4b3ece6c236d9dcR186-R188
Thank you for the review!

Reply:
Hi @kgyrtkirk, could you please take another look if have some secs?